docs/src/basics/blockchain_types.md

```diff
@@ -43,3 +43,35 @@ let my_number: b256 = 0x00000000000000000000000000000000000000000000000000000000
 let my_contract_id: ContractId = ~ContractId::from(my_number);
 let forty_two: b256 = my_contract_id.into();
 ```
+
+## `Identity` Type
+
+The `Identity` type is an enum that allows for the handling of both `Address` and `ContractId` types. This is useful in cases where either type is accepted, e.g. receiving funds from an identified sender, but not caring if the sender is an address or a contract.
+
+An `Identity` is implemented as follows.
+
+```sway
+{{#include ../../../sway-lib-std/src/identity.sw:7:10}}
```

Comment:
While I like the idea here, I worry these line number can get out of date for any number of reasons. We have the following options:
1. Keep it as is
2. Inline the code directly here
3. Use anchors in the standard library itself. 

If we're going to start documenting the standard library directly in the Sway Book (due to the concerns that @adlerjohn raised with docstrings in Sway code), we may have to do 3 eventually in other places. John, what do you think?

Reply:
Or maybe just have a link to the `identity.sw` file instead of having code in here.

Reply:
There's no issue, but only add them as needed.